setuptools/extension.py

```diff
@@ -126,6 +126,14 @@ class Extension(_Extension):
       specified on Windows. (since v63)
     """
 
+    # These 4 are set and used in setuptools/command/build_ext.py
+    # The lack of a default value and risk of `AttributeError` is purposeful
+    # to avoid people forgetting to call finalize_options if they modify the extension list
```

Comment:
```suggestion
    # to avoid people forgetting to call finalize_options if they modify the extension list.
    # See example/rationale in https://github.com/pypa/setuptools/issues/4529.
```